homeassistant/components/unifiprotect/event.py

```diff
@@ -0,0 +1,102 @@
+"""Platform providing event entities for UniFi Protect."""
+
+from __future__ import annotations
+
+import dataclasses
+
+from uiprotect.data import (
+    Camera,
+    EventType,
+    ProtectAdoptableDeviceModel,
+    ProtectModelWithId,
+)
+
+from homeassistant.components.event import (
+    EventDeviceClass,
+    EventEntity,
+    EventEntityDescription,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import ATTR_EVENT_ID
+from .data import ProtectData, UFPConfigEntry
+from .entity import EventEntityMixin, ProtectDeviceEntity
+from .models import ProtectEventMixin
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class ProtectEventEntityDescription(ProtectEventMixin, EventEntityDescription):
+    """Describes UniFi Protect event entity."""
+
+
+EVENT_DESCRIPTIONS: tuple[ProtectEventEntityDescription, ...] = (
+    ProtectEventEntityDescription(
+        key="doorbell",
+        translation_key="doorbell",
+        name="Doorbell",
+        device_class=EventDeviceClass.DOORBELL,
+        icon="mdi:doorbell-video",
+        ufp_required_field="feature_flags.is_doorbell",
+        ufp_event_obj="last_ring_event",
+        event_types=[EventType.RING],
+    ),
+)
+
+
+class ProtectDeviceEventEntity(EventEntityMixin, ProtectDeviceEntity, EventEntity):
+    """A UniFi Protect event entity."""
+
+    entity_description: ProtectEventEntityDescription
+
+    @callback
+    def _async_update_device_from_protect(self, device: ProtectModelWithId) -> None:
+        description = self.entity_description
+
+        prev_event = self._event
+        prev_event_end = self._event_end
+        super()._async_update_device_from_protect(device)
+        if event := description.get_event_obj(device):
+            self._event = event
+            self._event_end = event.end if event else None
+
+        if (
+            event
+            and not self._event_already_ended(prev_event, prev_event_end)
+            and (event_types := description.event_types)
+            and (event_type := event.type) in event_types
+        ):
+            self._trigger_event(event_type, {ATTR_EVENT_ID: event.id})
+            self.async_write_ha_state()
```

Comment:
Why wouldn't `_trigger_event` write the state inside event entity? 🤔

Reply:
I'm not sure why the `event` platform works that way.  Maybe someone would want to modify another _attr before writing?